src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -128,6 +128,9 @@ public void Add(X509Certificate2 certificate)
         {
             if (certificate == null)
                 throw new ArgumentNullException(nameof(certificate));
+            
+            if (certificate.Handle == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(certificate));
```

Comment:
Hmm, while it didn't show up in the test run I triggered, I ran similar code to the test on netfx:

```C#
            try
            {
                X509Store my = new X509Store(StoreName.My, StoreLocation.CurrentUser);
                my.Open(OpenFlags.ReadWrite);
                my.Add(new X509Certificate2());
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
            }
```

Resulting in

>System.Security.Cryptography.CryptographicException: pCertContext is an invalid handle.
   at System.Security.Cryptography.CAPI.CertAddCertificateContextToStore(SafeCertStoreHandle hCertStore, SafeCertContextHandle pCertContext, UInt32 dwAddDisposition, SafeCertContextHandle ppStoreContext)
   at System.Security.Cryptography.X509Certificates.X509Store.Add(X509Certificate2 certificate)